LanternMobileTestbed/app/src/main/java/org/lantern/lanternmobiletestbed/Browse.java

```diff
@@ -0,0 +1,141 @@
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    public void onToggleLantern(View view) {
+        getIPAddressView().setText("Toggling Lantern ...");
+        getIPAddressView().setEnabled(false);
+        ToggleButton button = (ToggleButton) view;
+        new AsyncTask<Boolean, Void, String>() {
+            @Override
+            protected String doInBackground(Boolean... params) {
+                boolean on = params[0];
+                try {
+                    if (on) {
+                        Log.i(TAG, "Turning on proxy");
+                        int startupTimeoutMillis = 30000;
+                        String trackingId = "UA-21815217-17";
+                        Lantern.enable(getApplicationContext(), startupTimeoutMillis, trackingId);
+                        Log.i(TAG, "Turned on proxy");
+                    } else {
+                        Log.i(TAG, "Turning off proxy");
+                        Lantern.disable();
+                        Log.i(TAG, "Turned off proxy");
+                    }
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+                return null;
+            }
+
+            @Override
+            protected void onPostExecute(String ipAddress) {
+                refreshIP(null);
+            }
+        }.execute(button.isChecked());
+    }
+
+    public void refreshIP(View view) {
+        getIPAddressView().setText("refreshing ...");
+        getIPAddressView().setEnabled(false);
+        new AsyncTask<Void, Void, String>() {
+            @Override
+            protected String doInBackground(Void... params) {
+                try {
+                    Log.i(TAG, "Opening connection to " + GEO_LOOKUP);
+                    URL url = new URL(GEO_LOOKUP);
+                    HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+                    // Need to force closing so that old connections (with old proxy settings) don't get reused.
+                    urlConnection.setRequestProperty("Connection", "close");
+                    try {
+                        InputStream in = new BufferedInputStream(urlConnection.getInputStream());
+                        Scanner s = new Scanner(in).useDelimiter("\\A");
+                        return s.hasNext() ? s.next() : "";
+                    } finally {
+                        urlConnection.disconnect();
+                        Log.i(TAG, "Finished doing geolookup");
+                    }
+                } catch (Exception e) {
+                    return "Unable to refresh IP: " + e.getMessage();
+                }
+            }
```

Comment:
This refreshIP test would be really useful to add to the connectivity checker, as a quick test to verify the status of the VPN connection. What do you think of factoring this out (and reusing it here of course) for that purpose? Maybe we could add it to ``Utils`` that comes with the SDK?